### ISSUE TITLE
Harmonic Analysis

### ISSUE THREAD

**Issue by lthurner** [e2nIEE/pandapower#1292]
Thanks to @tomislavantic for implementing a harmonic analysis for pandapower! This is a great new function for pandapower networks.

In order to integrate the developed functions smoothly into pandapower, I see a couple of  that need improving:

**Tests**
The testcases are based on excel files that are not included in the repo. It would be best to either save the test grids directly as .jsons or create the networks in the code without external excel files
Furthermore, all test cases should be included in the test suite as executable pytest cases: https://github.com/e2nIEE/pandapower/tree/harmonics/pandapower/test

**Validation**
There are multiple tests, but I don't see any validation yet (correct me if I am wrong?). Do we have calculation results from any other source (examples from literature, standards or other software tools) that we can compare the results calculated with the new pandapower function? Compared to how it is done for power flow here: https://github.com/e2nIEE/pandapower/blob/develop/pandapower/test/loadflow/test_results.py

**Result handling**
In pandapower, we have the convention that all results are saved in res_<element> tables. Right now, all results are returned from the analysis function. Harmonics results should be written to result tables as well.

**File Structure**
Minor point, but I think the module names could be a little shorter. I would suggest pandapower/harmonics/balanced and pandapower/harmonics/unbalanced. Also, the module should be importable (needs an __init__.py file, see other modules).

In summary, the goal should be that the calculation can be called like this:

```
import pandapower.harmonics as harmonic
harmonic.balanced_thd_voltage(net)
print(net.res_bus_harmonics)
```
